runtime/knative/knative.go

```diff
@@ -58,15 +60,20 @@ func (r *Runtime) RegisterOpenFunction(
 	ctx.InitDaprClientIfNil()
 
 	// Register the synchronous function (based on Knaitve runtime)
-	r.handler.HandleFunc(rf.GetPath(), func(w http.ResponseWriter, r *http.Request) {
+	route := r.handler.HandleFunc(rf.GetPath(), func(w http.ResponseWriter, r *http.Request) {
 		rm := runtime.NewRuntimeManager(ctx, prePlugins, postPlugins)
-		rm.FuncContext.SetSyncRequest(w, r)
+		// save the Vars into the context
+		_ctx := ofctx.CtxWithVars(r.Context(), ofctx.Vars(r))
+		rm.FuncContext.SetNativeContext(_ctx)
+		rm.FuncContext.SetSyncRequest(w, r.WithContext(_ctx))
 		defer RecoverPanicHTTP(w, "Function panic")
 		rm.FunctionRunWrapperWithHooks(rf.GetOpenFunctionFunction())
 
 		switch rm.FuncOut.GetCode() {
 		case ofctx.Success:
 			w.Header().Set(functionStatusHeader, successStatus)
+			w.WriteHeader(rm.FuncOut.GetCode())
+			w.Write(rm.FuncOut.GetData())
```

Comment:
@tpiperatgod do we need to `w.Write(rm.FuncOut.GetData())` here?
we don't have this previously

Reply:
need to refer to the mechanism of the ResponseWriter in the http package: https://pkg.go.dev/net/http#ResponseWriter